editoast/src/models/infra.rs

```diff
@@ -289,6 +289,45 @@ impl List<NoParams> for Infra {
     }
 }
 
+#[async_trait]
+impl Delete for Infra {
+    async fn delete_conn(conn: &mut PgConnection, infra_id: i64) -> Result<bool> {
+        use crate::tables::infra_object_track_section::dsl as track_section_dsl;
+        use crate::tables::search_track::dsl as search_track_dsl;
+
+        sql_query("ALTER TABLE infra_object_track_section DISABLE TRIGGER search_track__del_trig")
+            .execute(conn)
+            .await
+            .expect("Failed to disable trigger");
+        delete(
+            track_section_dsl::infra_object_track_section
+                .filter(track_section_dsl::infra_id.eq(infra_id)),
+        )
+        .execute(conn)
+        .await
+        .expect("Failed to delete from infra_object_track_section");
+        delete(
+            search_track_dsl::search_track.filter(search_track_dsl::infra_id.eq(infra_id as i32)),
+        )
+        .execute(conn)
+        .await
+        .expect("Failed to delete from search_track");
+        sql_query("ALTER TABLE infra_object_track_section ENABLE TRIGGER search_track__del_trig")
+            .execute(conn)
+            .await
+            .expect("Failed to enable trigger");
+
+        match delete(dsl::infra.filter(dsl::id.eq(infra_id)))
+            .execute(conn)
+            .await
+        {
+            Ok(1) => Ok(true),
+            Ok(_) => Ok(false),
+            Err(err) => Err(err.into()),
+        }
+    }
```

Comment:
I believe you should be able to write this whole block with the following tricks:
- return with `?` directly
- then only return the result of an equality test with the unwrapped value

```rust
delete(dsl::infra.filter(dsl::id.eq(infra_id))).execute(conn).await? == 1
```

Reply:
done ! 👍🏽